pkg/util/ecs/metadata/clients.go

```diff
@@ -37,6 +40,10 @@ type util struct {
 // endpoint, by detecting the endpoint address. Returns an error if it was not
 // possible to detect the endpoint address.
 func V1() (*v1.Client, error) {
+	if !config.IsCloudProviderEnabled(cloudProviderName) {
+		return nil, fmt.Errorf("cloud provider is disabled by configuration")
```

Comment:
maybe you can add the `cloudProviderName` in the error

Reply:
Makes sense, done 👍